src/runtime/providers/math.ts

```diff
@@ -24,10 +26,13 @@ abstract class TotalFunctionConstraint extends Constraint {
 
   // Total functions always have a cardinality of 1
   getProposal(tripleIndex, proposed, prefix) {
-    let proposal = this.proposalObject;
-    proposal.providing = proposed;
-    proposal.cardinality = 1;
-    return proposal;
+    if(this.returns.length) {
```

Comment:
To bring the code in line with the way that providers\logical.ts handles the same situation.
There's a slight discrepancy with the documentation in that it isn't returning a proposal with cardinality zero, but I've been unable to show that simply returning produces any strange behavious, so assumed returning an undefined is handled correctly by the runtime.